test/unit/traits/src/NativeHandleTest.cpp

```diff
@@ -0,0 +1,26 @@
+/* Copyright 2022 Antonio Di Pilato
+ *
+ * This file is part of alpaka.
+ *
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+
+#include <alpaka/acc/Traits.hpp>
+#include <alpaka/test/acc/TestAccs.hpp>
+#include <alpaka/traits/Traits.hpp>
+
+#include <catch2/catch.hpp>
+
+TEMPLATE_LIST_TEST_CASE("NativeHandle", "[handle]", alpaka::test::TestAccs)
+{
+    using Dev = alpaka::Dev<TestType>;
+    using Pltf = alpaka::Pltf<Dev>;
+    Dev const dev(alpaka::getDevByIdx<Pltf>(0u));
+    auto handle = alpaka::getNativeHandle(dev);
+
+    STATIC_REQUIRE(std::is_same_v<alpaka::NativeHandle<Dev>, decltype(handle)>);
+    STATIC_REQUIRE(std::is_same_v<alpaka::NativeHandle<Dev>, int>); // It won't work for SYCL backend
```

Comment:
What is this for?

Reply:
Testing the consistency of the alias Native Handle (same type of the handle returned by the getNativeHandle function) and that the NativeHandle of a device is an integer (but this will be changed with Sycl backend)

Reply:
First line tests `getNativeHandle` and third line tests consistency of the returned type with the trait.

The last line is to check that the native type is `int` which is true for all our backends at the moment. We will need to adapt this when the SYCL branch comes in, since it probably has a different native handle. I was not sure whether I want this last test initially, but this way we can enforce that we do not break the native handle type accidentally (e.g. making it suddently unsigned somewhere).